src/Builders/OrderBuilder.php

```diff
@@ -0,0 +1,239 @@
+          continue;
+        }
+
+        $orderLines->addLine(new Line(
+          type: 'shipping_fee',
+          merchantOrderLineId: 'shipping',
+          name: $shipment->getShippingMethod()->getName(),
+          quantity: 1,
+          amount: new Amount(floatval($shipment->getAmount()->getNumber()) * 100),
+          vatPercentage: new VatPercentage(new Percentage(0)),
+          currency: new Currency(
+            $shipment->getAmount()->getCurrencyCode()
+          )
+        ));
+      }
+      return $orderLines;
+    }
+  }
+
+  /**
+   * Get info about ordered products.
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *
+   * @return array
+   */
+  protected function getOrderItemsInfo(OrderInterface $order)
+  {
+    $order_items_info = [];
+    $order_items = $order->getItems();
+    foreach ($order_items as $order_item) {
+      if (!empty($order_item)) {
+        $order_items_info[] = [
+          'id' => $order_item->id(),
+          'quantity' => number_format($order_item->getQuantity()),
+          'name' => $order_item->getTitle(),
+          'price' => $order_item->getUnitPrice()->getNumber()
+        ];
+      }
+    }
+    return $order_items_info;
+  }
+
+  public function getWebhookUrl($payment)
+  {
+    /** @var \Drupal\commerce_payment\Plugin\Commerce\PaymentGateway\OffsitePaymentGatewayInterface $payment_gateway_plugin */
+    $payment_gateway_plugin = $payment->getPaymentGateway()->getPlugin();
+    return $payment_gateway_plugin->getNotifyUrl()->toString();
+  }
+
+
+  public function getReturnUrl($payment)
+  {
+    /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+    $order = $payment->getOrder();
+    return Url::fromRoute('commerce_payment.checkout.return', [
+      'commerce_order' => $order->id(),
+      'step' => 'payment',
+    ], ['absolute' => TRUE])->toString();
+  }
+
+  public function getCancelUrl($payment) {
+    /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+    $order = $payment->getOrder();
+
+    return Url::fromRoute('commerce_payment.checkout.cancel', [
+      'commerce_order' => $order->id(),
+      'step' => 'payment',
+    ], ['absolute' => TRUE])->toString();
+  }
+
+  public function createOrder(
+    PaymentInterface $payment,
+    Client $client,
+    Customer $customer,
+    $paymentMethod,
+    $issuerId = null,
+    $verifiedTerms = null
+  ) {
+    $payment_amount = $payment->getAmount();
+    $raw_amount = $payment_amount->getNumber();
+    $transactions = $this->prepareTransaction($paymentMethod, $this->preparePaymentMethodDetails($issuerId, $verifiedTerms));
+
+    $order = new Order(
+      currency:  new Currency($payment_amount->getCurrencyCode()),
+      amount: new Amount(floatval($raw_amount)*100),
+      transactions: $transactions,
+      customer: $customer,
+      orderLines: $this->getOrderLines($payment),
+      extra: new Extra(
+        $this->clientBuilder->getExtraLines()
```

Comment:
if we try to redefine getExtraLines() in BuilderRedefiner - it won't be redefined here, cause here we invoke this func using the object of clientBuilder.